services/server/src/wss/handlers/base.ts

```diff
@@ -19,7 +20,7 @@ export abstract class WssHandler {
     this.socket.broadcast.to(room).emit(event, ...data);
   }
 
-  revert(payload: Wss.RevertPayload) {
-    this.socket.emit(Wss.ServerEvent.Revert, payload);
+  protected must(callback: AcknowledgeCallback | undefined, payload: AcknowledgePayload): void {
+    return callback ? callback(payload) : console.error("Socket.io callback is not defined!");
   }
```

Comment:
No need to throw/log an error, The frontend might not be interested in cacthing errors.

```suggestion
  protected call(callback: AcknowledgeCallback | undefined, payload: AcknowledgePayload): void {
    if (!callback) return;
    return callback(payload);
  }
```

Reply:
This is for debugging in the backend... no thing to do with the frontned. Calling `this.must` or `this.call` without any reflection will be mysterious and puzzles the user (developer).

Reply:
From the first look, it is not clear that it is used for debugging. If you want to warn the developer not to do this, then you should throw an error instead.

Reply:
It's completely normal for callback to be undefined; in other word, it's not an error to not pass the callback parameter. That's why I considered using the log.

Reply:
Sounds good.